### PR TITLE
fix(building-webpack): support windows file pathes as well

### PR DESCRIPTION
**Pull request by daKmoR**
closes https://github.com/open-wc/open-wc/issues/342

nodes `path.join` uses system separators so on windows it will create urls like `foo\bar.js`... that maybe still works for browser if they are nice and accept/change them... but Terser just says `\b` is not a valid escape thing and breaks...

=> now we force `/` even on windows

### CODE REVIEW

File: packages/building-webpack/src/modern-web-webpack-plugin.js [hunk 1]
```diff
@@ -54,6 +54,11 @@ const legacyError = error(
   'You are running webpack-dev-server with a legacy build, run the dev server with the --legacy flag for this browser or run a production build.',
 );
 
+function pathJoin(...args) {
```
Comment (LarsDenBakker) [inline packages/building-webpack/src/modern-web-webpack-plugin.js]: I suggest naming this browserPathJoin or something like that. Also, why do we need to call apply here?

File: packages/building-webpack/src/modern-web-webpack-plugin.js [hunk 2]
```diff
@@ -54,6 +54,11 @@ const legacyError = error(
   'You are running webpack-dev-server with a legacy build, run the dev server with the --legacy flag for this browser or run a production build.',
```
Comment (LarsDenBakker) [inline packages/building-webpack/src/modern-web-webpack-plugin.js]: I think the solution is correct but I am not sure if we need to apply it in all places. I think we only need this in places where we ship to the browser, right? Any code that runs in mode still needs os specific separators. Are you sure we got this covered correctly?

**Comment by LarsDenBakker** [comment]
Great find, thanks!

File: packages/building-webpack/src/modern-web-webpack-plugin.js (hunk 2 above)
Comment (jorenbroekema) [reply packages/building-webpack/src/modern-web-webpack-plugin.js]: I tested this code on Windows 7 just now and it works ;).

Comment (daKmoR) [reply packages/building-webpack/src/modern-web-webpack-plugin.js]: all the pathes changed are only used to finally end up in the frontend... so I don't see a path where we should not have applied it?

File: packages/building-webpack/src/modern-web-webpack-plugin.js (hunk 1 above)
Comment (daKmoR) [reply packages/building-webpack/src/modern-web-webpack-plugin.js]: `path.join takes` a list of n argument. can you `forward` n arguments without using apply?

Comment (LarsDenBakker) [reply packages/building-webpack/src/modern-web-webpack-plugin.js]: Yeah I think you can just do `path.join(...args)`

File: packages/building-webpack/src/modern-web-webpack-plugin.js (hunk 2 above)
Comment (LarsDenBakker) [reply packages/building-webpack/src/modern-web-webpack-plugin.js]: Ok, thanks for testing. I was reviewing on my phone so it was hard to tell.

**Pull request merged**